test/basics.jl

```diff
@@ -0,0 +1,45 @@
+string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UniStr]
+
+@testset "constructors" begin
+    for strtype in string_types
+        @test convert(strtype, [0x61,0x62,0x63,0x21]) == "abc!"
+        @test convert(strtype, "abc!") == "abc!"
+
+        @test isempty(string())
```

Comment:
This tests nothing about the new string types.
Probably the best way would be to do the following:
`emptystr = convert(T, "")` (using T instead of strtype, as I suggested above)
then
`@test isempty(emptystr)`